bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/Observation.scala

```diff
@@ -1,36 +1,43 @@
 package edu.gemini.model.p1.immutable
 
+import edu.gemini.model.p1.overheads.Overheads
 import edu.gemini.model.p1.{mutable => M}
+
 import scalaz._
 import Scalaz._
 
 object Observation {
 
-  // Some lenses
+  // Some lenses.
   val blueprint:Lens[Observation, Option[BlueprintBase]] = Lens.lensu((a, b) => a.copy(blueprint = b), _.blueprint)
   val condition:Lens[Observation, Option[Condition]] = Lens.lensu((a, b) => a.copy(condition = b), _.condition)
   val target:Lens[Observation, Option[Target]] = Lens.lensu((a, b) => a.copy(target = b), _.target)
   val meta:Lens[Observation, Option[ObservationMeta]] = Lens.lensu((a, b) => a.copy(meta = b), _.meta)
-  val time:Lens[Observation, Option[TimeAmount]] = Lens.lensu((a, b) => a.copy(time = b), _.time)
+  val intTime:Lens[Observation, Option[TimeAmount]] = Lens.lensu((a, b) => a.copy(intTime = b), _.intTime)
```

Comment:
I wasn't sure about this. The only time `progTime` should change is as a consequence of `intTime`, so I didn't think we needed a `Lens`, since the only instance `progTime` will be directly set is through migration from an older proposal (as opposed to implicitly set by changing `intTime`).

So I'm not entirely certain if a `Lens` is needed, but if there is a reason to use one, I'm happy to add one.

I'm working on some tests, including an upconversion one. I think it's essential since from semester-to-semester, the `Overheads` could change according to my conversation with Bryan, which is also why I recalculate the calculated times from the integration time.

In the final PR (remember that this is not the final PR), I plan to include tests, and will solve the issue of Travis failing 😸